execution_engine/src/core/tracking_copy/mod.rs

```diff
@@ -462,6 +465,14 @@ impl<R: StateReader<Key, StoredValue>> StateReader<Key, StoredValue> for &Tracki
     ) -> Result<Option<TrieMerkleProof<Key, StoredValue>>, Self::Error> {
         self.reader.read_with_proof(correlation_id, key)
     }
+
+    fn read_trie(
```

Comment:
This will load the whole global state trie into memory? That can be gigabytes of data.

Reply:
No it won't

Reply:
How about you tell me a bit more about it? What will it load then? Can you be more helpful? From what I see, this returns a `Trie<K, V>` where `V=StoredValue`.

Reply:
How about you read the code?

Reply:
> How about you read the code?

Specifically: 

https://github.com/CasperLabs/casper-node/blob/d18c4b875ac319b9451940624d644a789b306e38/execution_engine/src/storage/trie/mod.rs#L251-L257

A `Trie` value is not fully-rendered.   Put alternately, this is not a recursive representation in the way you might be thinking.  

It merely loads a single node of the trie.